### PR TITLE
If session expires mid-experiment, redirect to login page

### PR DESCRIPTION
**Pull request by abought**
Refs: https://openscience.atlassian.net/browse/LEI-345

⚠️ Companion to, and must be merged after, https://github.com/CenterForOpenScience/exp-addons/pull/214

💡  Review note: The first commit is just reformatting; view individual commit diffs. It's more straightforward than the full diff looks.

## Purpose
If a user leaves their browser open too long, they will be logged out by the server, but the app was not reflecting this. This fixes a bug where the user was allowed to pass through the survey even though data wasn't being saved. Now if save fails, they will be sent to login page.

Leverage `DataAdapterMixin` capabilities to invalidate the session when appropriate, and trigger appropriate behavior. This app uses ESA in slightly unorthodox ways, so we need to manually turn on this functionality.

Borrows from: https://github.com/simplabs/ember-simple-auth/blob/1.2.0-beta.1/addon/mixins/application-route-mixin.js#L128-L135

## Summary of changes
- Make sure that both 401 and 403 responses trigger session invalidation
- When a session is validated during the ISP survey, then when the user clicks next they should be kicked back to the login url
- For this special case: since data cannot be saved, do not prompt the user about unsaved data. Simply kick them out immediately.

## Testing notes
See JIRA ticket.

### CODE REVIEW

**Comment by abought** [comment]
Reminder to self: update submodule commit before merging

File: app/routes/participate/survey.js [hunk 1]
```diff
@@ -6,10 +6,21 @@ import config from 'ember-get-config';
...
+    session: Ember.inject.service(),
 
+    init() {
+        // If any request fails due to authentication reasons, kick user to login page
+        // Selectively adapted from:
+        //  https://github.com/simplabs/ember-simple-auth/blob/1.2.0-beta.1/addon/mixins/application-route-mixin.js#L113
+        this.get('session').on('invalidationSucceeded', Ember.run.bind(this, () => {
+            // Since the user has work in progress, we must turn off the prompt-on-exit behavior.
+            Ember.$(window).off('beforeunload');
+            window.location.replace('');
```
Comment (binoculars) [inline app/routes/participate/survey.js:21]: Why `replace` instead of `reload`?
http://www.w3schools.com/jsref/met_loc_reload.asp

Comment (abought) [reply app/routes/participate/survey.js:21]: Good questions!

Two main reasons:
1. We don't want to reload and be back at the survey page; we consciously want to kick the user back to the landing URL (login)
2. Parity with Ember-simple-auth. Although this app's requirements are a bit too weird to use the functionality directly, this fix largely parallels what the auth framework would normally try to do for free.  https://github.com/simplabs/ember-simple-auth/blob/1.2.0-beta.1/addon/mixins/application-route-mixin.js#L133

Comment (binoculars) [reply app/routes/participate/survey.js:21]: 👍

**Review by binoculars — APPROVED** [review]
Small change, looks legit. Hard to test a session expiration, but as long as waited (or modified the session timeout) to test it and it works, then I say it's ready merge

**Comment by abought** [comment]
Yeah, I've been forcing pseudo-expiration using the method outlined in the PR description. But I agree that a REAL expiration is also worth testing, eg by leaving window open overnight.

**Pull request merged**